include/appbase/application_instance.hpp

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+namespace appbase {
+
+   using executor_t = appbase_executor;
+
+   // ------------------------------------------------------------------------------------------
+   class application : public application_base {
+   public:
+      static application&  instance() {
+         if (__builtin_expect(!!app_instance, 1))
+            return *app_instance;
+         app_instance.reset(new application);
+         return *app_instance;
+      }
+
+      static void reset_app_singleton() { app_instance.reset(); }
+      
+      static bool null_app_singleton()  { return !app_instance; }
+
+      template <typename Func>
+      auto post( int priority, Func&& func ) {
+         return application_base::post(executor_, priority, std::forward<Func>(func));
+      }
```

Comment:
I think this should be removed (or at least deprecated) since the interface will be different for multi-queue posting.

Reply:
OK, I have deprecated it!